sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -557,6 +581,32 @@ case class Sin(child: Expression) extends UnaryMathExpression(math.sin, "SIN")
   """)
 case class Sinh(child: Expression) extends UnaryMathExpression(math.sinh, "SINH")
 
+@ExpressionDescription(
+  usage = """
+    _FUNC_(expr) - Returns inverse hyperbolic sine of `expr`.
+  """,
+  arguments = """
+    Arguments:
+      * expr - hyperbolic angle
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_(0);
+       0.0
+  """,
+  since = "3.0.0")
+case class Asinh(child: Expression)
+  extends UnaryMathExpression((x: Double) => x match {
+    case Double.NegativeInfinity => Double.NegativeInfinity
+    case _ => math.log(x + math.sqrt(x * x + 1.0)) }, "ASINH") {
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    defineCodeGen(ctx, ev, c =>
+      s"""${ev.value} = \"$c\" == \"Double.NEGATIVE_INFINITY\" ? """ +
+                        s"java.lang.Double.NEGATIVE_INFINITY : " +
```

Comment:
nit: you can avoid `java.lang` I think

Reply:
all other codeGen in mathExpressions.scala keep `java.lang` so i'm doing it the same way, at least for `java.lang.Math`. @dongjoon-hyun what is the best practice in this case?

Reply:
yes I mean for `Double`. In `CodeGenerator` as boxed type we return just `Double`, so it would be great to be coherent